### PR TITLE
[API-1008]ILogger usage and message strings refactored

### PR DESCRIPTION
**Pull request by emreyigit**
All debug and warning log statements are refactored. 

- IfDebug and IfWarning extensions are used. -> Prevented unnecessary logger calls. 
- String interpolations are changed to parameter passing to logger method. -> Prevented unnecessary calculations for not logged messages.

I did some benchmark on the differences but I can't say that result is obvious since measurements are bound to client. Isolating the benchmarks requires amount of work, so I skipped to deep dive. However, in long term, there is no doubt that the refactoring will contribute to performance.

### CODE REVIEW

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        {
+            var map = await _client.GetMapAsync<string, string>(_mapName);
+            await map.PutAsync(_key, _value);
+        }
+
+        [Benchmark]
+        public async Task DoGet()
+        {
+            var map = await _client.GetMapAsync<string, string>(_mapName);
+            var readValue = await map.GetAsync(_key);
```
Comment (alisengul53) [inline src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs:95]: Do we have a chance to get the map inside `GlobalSetup`?
So in benchmark methods only put/get operations will be remained

Comment (alisengul53) [inline src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs:95]: Do we have a chance to get the map inside `GlobalSetup`?
So in benchmark methods only put/get operations will be remained

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            string[] words = new string[count];
+
+            for (int i = 0; i < count; i++)
+            {
+                string word = GenerateWord(minLength, maxLength);
+                words[i] = word;
+            }
+
+            return words;
+        }
```
Comment (alisengul53) [inline src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs:40]: Not used I think, are we planning to use it in the future?

File: src/Hazelcast.Net.Benchmarks/LoggerMessageTemplateVsEnabledCheck.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+
+        [Benchmark]
+        public void LogWithMsgTemplateCheck()
+        {
+            string msg = "we are sailing to Key Largo, goodluck";
+
+            _logger.LogDebug("something went wrong {msg}", msg);
+        }
```
Comment (alisengul53) [inline src/Hazelcast.Net.Benchmarks/LoggerMessageTemplateVsEnabledCheck.cs:45]: What about defining msg string globally?

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs (hunk 1 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs:95]: Hold on - how come this is part of the PR about logging at all?!

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs (hunk 2 above)
Comment (zpqrtbnk) [reply src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs:40]: Hold on - how come this is part of the PR about logging at all?!

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        //They does not work inline process. To overcome that
+        //seperate the class into a different project.
+        //details : https://benchmarkdotnet.org/articles/features/etwprofiler.html
+        //[NativeMemoryProfiler]
+        //[MemoryDiagnoser]
+        //[HardwareCounters(HardwareCounter.TotalCycles, HardwareCounter.BranchInstructions)]
+        public BasicMapOperation()
+        {
+
+        }
```
[the remark this answers is not in the document]
Comment (zpqrtbnk) [reply src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs:27]: Hold on - how come this is part of the PR about logging at all?!

**Comment by zpqrtbnk** [comment]
Need to rebase the PR on master and probably update more files ...

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs (hunk 1 above)
Comment (emreyigit) [reply src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs]: That was a left over, fixed at c5955e9a36087d0b645b845cda4133779c13a6ef

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs (hunk 2 above)
Comment (emreyigit) [reply src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs]: That was a left over, fixed at c5955e9a36087d0b645b845cda4133779c13a6ef

File: src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (emreyigit) [reply src/Hazelcast.Net.Benchmarks/BasicMapOperation.cs]: That was a left over, fixed at c5955e9a36087d0b645b845cda4133779c13a6ef

File: src/Hazelcast.Net.Benchmarks/LoggerMessageTemplateVsEnabledCheck.cs (hunk 3 above)
Comment (emreyigit) [reply src/Hazelcast.Net.Benchmarks/LoggerMessageTemplateVsEnabledCheck.cs]: Makes sense, fixed at c5955e9a36087d0b645b845cda4133779c13a6ef

**Comment by emreyigit** [comment]
Synced the branch with upstream. cc: @zpqrtbnk @alisengul53

**Review by zpqrtbnk — APPROVED** [review]
OK

**Review by alisengul53 — APPROVED** [review]
Looks good

**Pull request merged**